_releaser/aws.go

```diff
@@ -84,6 +96,189 @@ func (s *AwsLambdaInvokeCmd) Run() error {
+	// the lambda function code image is never ready right away, AWS has to
+	// process it, so we wait 3 seconds before trying to publish the version.
+	time.Sleep(3 * time.Second)
+
+	publishConfig, err := svc.PublishVersion(&lambda.PublishVersionInput{
+		FunctionName: aws.String(s.Function),
+		CodeSha256:   aws.String(*updateConfig.CodeSha256),
+		Description:  aws.String(ver),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to publish lambda function version %q for %q: %w", ver, s.Function, err)
+	}
+	log.Printf("INFO: lambda function version %q published successfully (%s)\n", ver, *publishConfig.FunctionArn)
+
+	sess, err := session.NewSession(&aws.Config{
+		Credentials: awsCredentials(),
+		Region:      aws.String(s.Region)},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create session: %w", err)
+	}
+
+	cfrt := cloudfront.New(sess)
+	cfrtDistrib, err := cfrt.GetDistribution(&cloudfront.GetDistributionInput{
+		Id: aws.String(s.CloudfrontID),
+	})
+	if err != nil {
+		return fmt.Errorf("cannot find cloudfront distribution %q: %w", s.CloudfrontID, err)
+	}
+	log.Printf("INFO: cloudfront distribution %q loaded\n", *cfrtDistrib.Distribution.Id)
+
+	cfrtDistribConfig, err := cfrt.GetDistributionConfig(&cloudfront.GetDistributionConfigInput{
+		Id: aws.String(s.CloudfrontID),
+	})
+	if err != nil {
+		return fmt.Errorf("cannot load cloudfront distribution config: %w", err)
+	}
+	log.Printf("INFO: cloudfront distribution configuration loaded\n")
+
+	distribConfig := cfrtDistribConfig.DistributionConfig
+	if distribConfig.DefaultCacheBehavior == nil {
+		log.Printf("INFO: cloudfront distribution default cache behavior not found. skipping...")
+		return nil
+	}
+
+	for _, funcAssoc := range distribConfig.DefaultCacheBehavior.LambdaFunctionAssociations.Items {
+		if *funcAssoc.EventType != cloudfront.EventTypeViewerRequest {
+			continue
+		}
+		log.Printf("INFO: cloudfront distribution viewer request function ARN found: %q\n", *funcAssoc.LambdaFunctionARN)
+	}
+
+	log.Printf("INFO: updating cloudfront config with viewer request function ARN %q", *publishConfig.FunctionArn)
+	distribConfig.DefaultCacheBehavior.LambdaFunctionAssociations = &cloudfront.LambdaFunctionAssociations{
+		Quantity: aws.Int64(1),
+		Items: []*cloudfront.LambdaFunctionAssociation{
+			{
+				EventType:         aws.String(cloudfront.EventTypeViewerRequest),
+				IncludeBody:       aws.Bool(false),
+				LambdaFunctionARN: publishConfig.FunctionArn,
+			},
+		},
+	}
+	_, err = cfrt.UpdateDistribution(&cloudfront.UpdateDistributionInput{
+		Id:                 aws.String(s.CloudfrontID),
+		IfMatch:            cfrtDistrib.ETag,
+		DistributionConfig: distribConfig,
+	})
+	if err != nil {
+		return err
+	}
+
+	log.Printf("INFO: cloudfront config updated successfully\n")
+	return nil
+}
+
+func getLambdaFunctionZip(funcFilename string, redirectsJSON string, redirectsPrefixesJSON string) ([]byte, error) {
+	funcdt, err := ioutil.ReadFile(funcFilename)
```

Comment:
Should be `os.` now;

```suggestion
	funcdt, err := os.ReadFile(funcFilename)
```

Reply:
There's one more (existing one) at line 36